packages/core/src/spawn.ts

```diff
@@ -25,21 +25,23 @@ export function createSpawner<
 
       if (behaviorCreator) {
         const resolvedName = name ?? 'anon'; // TODO: better name
-        const createdBehavior = behaviorCreator(context, _event.data, {
-          id: name || 'anon',
-          src: { type: behavior },
-          _event,
-          meta: undefined
-        });
+        const createdBehavior = isBehavior(behaviorCreator)
+          ? behaviorCreator
+          : behaviorCreator(context, _event.data, {
+              id: name || 'anon',
+              src: { type: behavior },
+              _event,
+              meta: undefined
+            });
```

Comment:
a simpler (more lightweight) variant:
```suggestion
        const createdBehavior =
          typeof behaviorCreator === 'function'
            ? behaviorCreator(context, _event.data, {
                id: name || 'anon',
                src: { type: behavior },
                _event,
                meta: undefined
              })
            : behaviorCreator;
```